blog/2015/04/how-to-contribute-with-template.html

```diff
@@ -0,0 +1,104 @@
+<h2 id="hacking">Hacking</h2>
+<p>For this how-to, suppose that I want to add the class <code>img-responsive</code> to every <code>img</code> elements of our lessons. One <strong>quickly and dirty</strong> way to do that is change <code>Makefile</code> with</p>
+<pre><code>$ git diff
+diff --git a/Makefile b/Makefile
+index 4d38850..ff49ca9 100644
+--- a/Makefile
++++ b/Makefile
+@@ -55,6 +55,7 @@ motivation.html : motivation.md _layouts/slides.revealjs Makefile
+        --filter=tools/filters/id4glossary.py \
+        $(INCLUDES) \
+        -o $@ $&lt;
++       sed -i &#39;s/^&lt;img /&lt;img class=&quot;img-responsive&quot; /g&#39; $@
+
+ # Pattern to convert R Markdown to Markdown.
+ %.md: %.Rmd $(R_CHUNK_OPTS)</code></pre>
+<p>To avoid losing my work, I save it in the staging area:</p>
+<pre><code>$ git add -u</code></pre>
+<p>Now I compile the HTML files:</p>
+<pre><code>$ make -B preview</code></pre>
+<p>And verify the changes.</p>
+<pre><code>$ git diff
+diff --git a/01-one.html b/01-one.html
+index 7622e55..1fc0d45 100644
+--- a/01-one.html
++++ b/01-one.html
+@@ -56,7 +56,8 @@ program&lt;/code&gt;&lt;/pre&gt;
+ &lt;/div&gt;
+ &lt;p&gt;or an image from the &lt;code&gt;figs&lt;/code&gt; directory:&lt;/p&gt;
+ &lt;div class=&quot;figure&quot;&gt;
+-&lt;img src=&quot;fig/example.svg&quot; title=&quot;this is the image&amp;#39;s alt text&quot; alt=&quot;this is the image&amp;#39;s title&quot; /&gt;&lt;p class=&quot;caption&quot;&gt;this is the image&#39;s title&lt;/p&gt;
++&lt;img class=&quot;img-responsive&quot; src=&quot;fig/example.svg&quot; title=&quot;this is the image&amp;#39;s alt text&quot; alt=&quot;this is the image&amp;#39;s title&quot; /&gt;
++&lt;p class=&quot;caption&quot;&gt;this is the image&#39;s title&lt;/p&gt;
+ &lt;/div&gt;
+ &lt;div id=&quot;challenge-title&quot; class=&quot;challenge panel panel-success&quot;&gt;
+ &lt;div class=&quot;panel-heading&quot;&gt;</code></pre>
+<p>This is exactly what I want. Great!!! I can now commit my changes.</p>
+<pre><code>$ git commit -m &#39;Add img-responsive class to img&#39;
+[example-centering-image a9cc86d] Add img-responsive class to img
+ 1 file changed, 1 insertion(+)</code></pre>
+<p>I will also save the change to the HTML file.</p>
+<pre><code>$ git add -u
+$ git commit -m &#39;Update HTML&#39;
+[example-centering-image 1ebf8c3] Update HTML
+ 1 file changed, 2 insertions(+), 1 deletion(-)</code></pre>
+<h2 id="copying-the-change-to-template">Copying the change to template</h2>
+<p>If you have taken the Git lesson at a Software Carpentry workshop, you should be comfortable with what you see until now. What I need to do is copy my change at <code>Makefile</code> to <a href="https://github.com/swcarpentry/lesson-template">template</a> using <code>git cherry-pick</code>. If you are trying to remember it from our workshop, don't panic because we don't teach it but you will discovery that it is very simple to use.</p>
```

Comment:
The explanation of how to do cherry picking is great, but I think it would be helpful to explain the why: why do you need to cherry pick this particular commit? and also the "what": what does cherry picking do?